src/main/java/org/gridsuite/study/server/service/StudyService.java

```diff
@@ -1181,23 +1187,23 @@ public String getNeworkAreaDiagram(UUID studyUuid, UUID nodeUuid, List<String> v
     }
 
     public void invalidateSecurityAnalysisStatusOnAllNodes(UUID studyUuid) {
-        securityAnalysisService.invalidateSaStatus(networkModificationTreeService.getStudySecurityAnalysisResultUuids(studyUuid));
+        securityAnalysisService.invalidateSaStatus(networkModificationTreeService.getComputationResultUuids(studyUuid, SECURITY_ANALYSIS));
```

Comment:
another factorization possibility...

Reply:
Maybe we could do this in another PR later...

Reply:
Fine, I will do it :) . And what a surprise : it leads to even more refactoring tied to this function !

Reply:
Changed my mind : All the computations use their own services. I don't think there is much to refactorize here. (unless the various services classes are refactored themselves, inheriting from a new parent abstract class ?)

Reply:
true it's ok for me.